listenbrainz/db/stats.py

```diff
@@ -45,29 +65,69 @@ def insert_user_stats(user_id, artists, recordings, releases, artist_count):
         )
 
 
-def get_user_stats(user_id):
-    """Get user stats for user with given ID.
+def get_user_stats(user_id, columns):
+    """ Get a particular stat for user with the given row ID.
 
-        Args: user_id (int): the row ID of the user in the DB
+        Args:
+            user_id (int): the row ID of the user in the DB
+            columns (str): the column name(s) of the user stat to be retrieved
 
-        Returns: A dict of the following format
-                 {
-                    "user_id" (int): the id of the user
-                    "artists" (dict): artist stats for the user
-                    "releases" (dict) : release stats for the user
-                    "recordings" (dict): recording stats for the user
-                    "last_updated" (datetime): timestamp when the stats were last updated
-                 }
+        Returns:
+            A dict of the following format
+            {
+                'user_id' (int): the row ID of the user in the DB,
+                '<stat>'  (dict): the stats requested
+                'last_updated' (datetime): datetime object representing when
+                                           this stat was last updated
+            }
     """
-
     with db.engine.connect() as connection:
         result = connection.execute(sqlalchemy.text("""
-            SELECT user_id, artist, release, recording, last_updated
+            SELECT user_id, {columns}, last_updated
               FROM statistics.user
              WHERE user_id = :user_id
-            """), {
+            """.format(columns=columns)), {
                 'user_id': user_id
             }
         )
         row = result.fetchone()
         return dict(row) if row else None
+
+
+def get_user_artists(user_id):
+    """Get top artists for user with given ID.
+
+        Args:
+            user_id (int): the row ID of the user in the DB
+
+        Returns:
+            A dict of the following format
+            {
+                'user_id' (int): the row ID of the user in the DB,
+                'artist'  (dict): artist stats for the user
+                'last_updated' (datetime): datetime object representing when
+                                        this stat was last updated
+            }
+    """
+    return get_user_stats(user_id, 'artist')
+
+
+def get_all_user_stats(user_id):
+    """ Get ALL user stats for user with given ID.
+
+        Args:
+            user_id (int): the row ID of the user in the DB
+
+        Returns:
+            A dict of the following format
+            {
+                'user_id' (int): the row ID of the user in the DB
+                'artist' (dict): artist stats for the user
```

Comment:
Same here -- if they all fit a common pattern, maybe describe this pattern somewhere?